python/utilities/dnslookup/dnslookup.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+
+# Implement DNS inclusion proof checking, see [TBD].
+#
+# Unfortunately, getting at the SCTs in general is hard in Python, so this
+# does not start with an SSL connection, but instead fetches a log entry by
+# index and then verifies the proof over DNS.
+
+# You will need to install DNSPython (http://www.dnspython.org/)
+
+import base64
+import dns.resolver
+import hashlib
+import json
+import logging
+import os
+import subprocess
+import sys
+import tempfile
+import urllib2
+
+ALL_LOGS_URL = 'https://www.certificate-transparency.org/known-logs/all_logs_list.json'
+
+basepath = os.path.dirname(sys.argv[0])
+sys.path.append(os.path.join(basepath, '../../../python'))
+from ct.crypto import merkle, verify
+from ct.proto import ct_pb2, client_pb2
+
+ctpath = os.path.join(basepath, '../../../cpp/client/ct')
+libpath = os.path.join(basepath, '../../../../install/lib')
+
+class CTDNSLookup:
+    def __init__(self, domain, verifier, nameservers, port = 53):
+        self.verifier = verifier
+        self.domain = domain
+        self.resolver = dns.resolver.get_default_resolver()
+        self.resolver.nameservers = nameservers
+        self.resolver.port = port
```

Comment:
Getting the default resolver, then overriding its nameservers and port, does that make any sense?

I'd have expected something like `resolver=None` in the parameters, and then this:

```python
self.resolver = resolver
if not self.resolver:
  self.resolver = dns.resolver.get_default_resolver()
```